lib/dotcom_web/components/trip_planner/itinerary_group.ex

```diff
@@ -2,47 +2,146 @@ defmodule DotcomWeb.Components.TripPlanner.ItineraryGroup do
+          <.icon :if={index > 0} name="angle-right" class="font-black w-2" />
+          <.leg_icon {summary_leg} />
+        <% end %>
+      </div>
+      <div class="flex flex-wrap gap-1 items-center mb-3 text-sm text-grey-dark">
+        <div :if={@summary.accessible?} class="inline-flex items-center gap-0.5">
+          <.icon type="icon-svg" name="icon-accessible-small" class="h-3 w-3 mr-0.5" /> Accessible
+          <.icon name="circle" class="h-0.5 w-0.5 mx-1" />
+        </div>
+        <div class="inline-flex items-center gap-0.5">
+          <.icon name="person-walking" class="h-3 w-3" />
+          <%= @summary.walk_distance %> mi
+        </div>
+        <div :if={@summary.total_cost > 0} class="inline-flex items-center gap-0.5">
+          <.icon name="circle" class="h-0.5 w-0.5 mx-1" />
+          <.icon name="wallet" class="h-3 w-3" />
+          <%= Fares.Format.price(@summary.total_cost) %>
+        </div>
+      </div>
+      <div class="flex justify-end items-center">
+        <div :if={Enum.count(@summary.next_starts) > 0} class="grow text-sm text-grey-dark">
+          Similar trips depart at <%= Enum.map(@summary.next_starts, &format_datetime_short/1)
+          |> Enum.join(", ") %>
+        </div>
+        <button class="btn-link font-semibold underline" phx-click={JS.toggle(to: "##{@group_id}")}>
+          Details
+        </button>
+      </div>
+      <div id={@group_id} class="mt-30" style="display: none;">
+        <.itinerary_detail :for={itinerary <- @itineraries} itinerary={itinerary} />
+      </div>
     </div>
     """
   end
 
+  attr(:class, :string, default: "")
+  attr(:routes, :list, required: true, doc: "List of %Routes.Route{}")
+  attr(:walk_minutes, :integer, required: true)
+
+  # No routes
+  defp leg_icon(%{routes: [], walk_minutes: _} = assigns) do
+    ~H"""
+    <span class={[
+      "flex items-center gap-1 text-sm font-semibold leading-none whitespace-nowrap py-1 px-2 rounded-full border border-solid border-gray-light",
+      @class
+    ]}>
+      <.icon name="person-walking" class="h-4 w-4" />
+      <span><%= @walk_minutes %> min</span>
+    </span>
+    """
+  end
+
+  # Group of commuter rail routes are summarized to one symbol.
+  defp leg_icon(%{routes: [%Routes.Route{type: 2} | _]} = assigns) do
+    ~H"""
+    <.route_symbol route={List.first(@routes)} class={@class} />
+    """
+  end
+
+  # No grouping when there's only one route!
+  defp leg_icon(%{routes: [%Routes.Route{}]} = assigns) do
+    ~H"""
+    <.route_symbol route={List.first(@routes)} {assigns} />
+    """
+  end
+
+  defp leg_icon(
+         %{routes: [%Routes.Route{type: type, external_agency_name: agency} | _]} = assigns
+       ) do
+    slashed? = type == 3 && is_nil(agency)
+
+    assigns =
+      assigns
+      |> assign(:slashed?, slashed?)
+      |> assign(
+        :grouped_classes,
+        if(slashed?,
+          do: "[&:not(:first-child)]:rounded-l-none [&:not(:last-child)]:rounded-r-none !px-3",
```

Comment:
<img width="574" alt="Screenshot 2024-11-15 at 5 06 22 PM" src="https://github.com/user-attachments/assets/b467454e-9622-4459-8d3c-6b13e8ee647f">
<img width="565" alt="Screenshot 2024-11-15 at 5 15 06 PM" src="https://github.com/user-attachments/assets/ee26bddb-ceb3-4a2d-b44a-99c319a61a06">


Something is going wonky with respect to the spacing. It looks like for whatever reason, the wrapping `div` thinks that each bus pill is a bit larger than it is (hence the fact that the weird empty space to the right is bigger when there are three routes versus when there are two).

What's especially weird is that it looks like this is only happening in Firefox. It looks great in Chrome.